### PR TITLE
Ensure nef file fit in vm

### PR DESCRIPTION
**Pull request by shargon**
Following the line of https://github.com/neo-project/neo-vm/pull/514

### CODE REVIEW

File: tests/Neo.UnitTests/SmartContract/UT_InteropService.NEO.cs [hunk 1]
```diff
@@ -121,12 +121,14 @@ public void TestContract_Create()
...
             var script_exceedMaxLength = new NefFile()
             {
-                Script = new byte[NefFile.MaxScriptLength - 1],
+                Script = new byte[ExecutionEngineLimits.Default.MaxItemSize - 50],
                 Source = string.Empty,
                 Compiler = "",
-                Tokens = System.Array.Empty<MethodToken>()
+                Tokens = Array.Empty<MethodToken>()
             };
-            script_exceedMaxLength.CheckSum = NefFile.ComputeChecksum(nef);
```
Comment (shargon) [inline tests/Neo.UnitTests/SmartContract/UT_InteropService.NEO.cs:129]: Note: This checksum was incorrectly computed

Comment (Liaojinghui) [reply tests/Neo.UnitTests/SmartContract/UT_InteropService.NEO.cs:129]: What this means, you have corrected the checksum? will this cause inconstancy?

Comment (shargon) [reply tests/Neo.UnitTests/SmartContract/UT_InteropService.NEO.cs:129]: No, is just a UT but it was wrong

**Review by shargon — COMMENTED** [review]
@Liaojinghui maybe we can just remove the constant and do this check in neo-devpack, otherwise, if it's not serialized/deserialized it won't happened. What do you think?

**Comment by Liaojinghui** [comment]
> @Liaojinghui maybe we can just remove the constant and do this check in neo-devpack, otherwise, if it's not serialized/deserialized it won't happened. What do you think?

I dont think it matters, bug you can do it and i will support you.

**Comment by AnnaShaleva** [comment]
> @Liaojinghui maybe we can just remove the constant and do this check in neo-devpack, otherwise, if it's not serialized/deserialized it won't happened.

Could you, please, clarify the constant name you're talking about? If it's the `ExecutionEngineLimits.Default.MaxItemSize`, then I don't think we need to remove this check from the core library, because native Management contract uses this deserialisation code on contract deploy/update, and without this check presented in the core library we can still create an unwantedly large contract.

**Comment by shargon** [comment]
> > @Liaojinghui maybe we can just remove the constant and do this check in neo-devpack, otherwise, if it's not serialized/deserialized it won't happened.
> 
> @shargon, could you, please, clarify the constant name you're talking about? If it's the `ExecutionEngineLimits.Default.MaxItemSize`, then I don't think we need to remove this check from the core library, because native Management contract uses this deserialisation code on contract deploy/update, and without this check presented in the core library we can still create an unwantedly large contract.

MaxScriptSize

**Comment by AnnaShaleva** [comment]
> maybe we can just remove the constant and do this check in neo-devpack

@shargon, I think it's a good idea to check the generated NEF file in the compiler against the size restrictions. I've implemented it in NeoGo, please, check the last two commits of https://github.com/nspcc-dev/neo-go/pull/3186/commits. If you're proposing something like this, then I support.

**Review by AnnaShaleva — APPROVED** [review]

**Review by Liaojinghui — APPROVED** [review]

**Comment by shargon** [comment]
> > maybe we can just remove the constant and do this check in neo-devpack
> 
> @shargon, I think it's a good idea to check the generated NEF file in the compiler against the size restrictions. I've implemented it in NeoGo, please, check the last two commits of https://github.com/nspcc-dev/neo-go/pull/3186/commits. If you're proposing something like this, then I support.

Yes, is like this

**Pull request merged**